build.vsh

```diff
@@ -31,39 +24,27 @@ enum ReleaseMode {
 	dev
 }
 
-fn (m ReleaseMode) cmd() fn () os.Result {
-	return match m {
-		.release { release }
-		.debug { debug }
-		.dev { dev }
+fn (m ReleaseMode) cc_flags() string {
+	$if windows {
+		return '-cc gcc' // TCC cannot build tree-sitter on Windows
+	} $else $if cross_compile_macos_arm64 ? {
+		return '-cc clang -cflags "-target arm64-apple-darwin"'
+	} $else {
+		return if m == .release { '-cc gcc' } else { '' }
 	}
 }
 
 fn errorln(msg string) {
 	eprintln('${term.red('[ERROR]')} ${msg}')
 }
 
-// debug builds the v-analyzer binary in debug mode.
-// This is the default mode.
-// Thanks to -d use_libbacktrace, the binary will print beautiful stack traces,
-// which is very useful for debugging.
-fn debug() os.Result {
-	libbacktrace := $if windows { '' } $else { '-d use_libbacktrace' }
-	return os.execute('${base_build_command} ${compiler_flag} -g ${libbacktrace}')
-}
-
-// dev builds the v-analyzer binary in development mode.
-// In this mode, additional development features are enabled.
-fn dev() os.Result {
+fn (m ReleaseMode) compile() os.Result {
 	libbacktrace := $if windows { '' } $else { '-d use_libbacktrace' }
-	return os.execute('${base_build_command} ${compiler_flag} -d show_ast_on_hover -g ${libbacktrace}')
-}
-
-// release builds the v-analyzer binary in release mode.
-// This is the recommended mode for production use.
-// It is about 30-40% faster than debug mode.
-fn release() os.Result {
-	return os.execute('${base_build_command} ${compiler_flag} -w -cflags "-O3 -DNDEBUG" -prod')
+	return match m {
+		.release { os.execute('${base_build_command} ${m.cc_flags()} -w -cflags "-O3 -DNDEBUG" -prod') }
+		.debug { os.execute('${base_build_command} ${m.cc_flags()} -w -cflags "-O3 -DNDEBUG" -prod') }
```

Comment:
debug should not have -prod it should have -g instead, it also should not have `-cflags "-O3 -DNDEBUG"` (those are implied by -prod)

Reply:
Agree. I kept the commands as they were, except that the prod build for linux has the `--cc gcc` flag

Reply:
also made an errors here. Thanks for the review. I'll close in favour splitting this